### PR TITLE
Instrument all controller actions

### PR DESCRIPTION
**Pull request by chancefeick**
Add support to instrument all controller actions, e.g., `instrument_action :all`. To enable instrumentation for all actions across all controllers, simply drop it in `ApplicationController`:

```ruby
class ApplicationController < ActionController::Base
  # Prevent CSRF attacks by raising an exception.
  # For APIs, you may want to use :null_session instead.
  protect_from_forgery with: :exception
  instrument_action :all
end
```

Originally introduced in librato/librato-rails#94.

@nextmat @bdehamer @obfuscurity

### CODE REVIEW

**Comment by nextmat** [comment]
Just want to verify that this inherits down from parent controllers, maybe add a test for that? Otherwise LGTM.

**Comment by ketiko** [comment]
Bump.  This would be really great to have.

**Comment by chancefeick** [comment]
Following up on this. Original implementation did not support inherited controllers, descendants were not loaded yet when subscribed. Took a pass at subscribing using `after_initialize` callback in `Railtie`. @nextmat @bdehamer, feedback appreciated.

File: README.md [hunk 1]
```diff
@@ -255,7 +255,6 @@ Symbols can be used interchangably with strings for metric names.
 `librato-rails` also has special helpers which are available inside your controllers:
 
 #### instrument_action
-_experimental_, this interface may change:
```
Comment (nextmat) [inline README.md]: I think we may want to leave this in there, although this has been around a while I still think we may change this interface some in future versions.

**Comment by nextmat** [comment]
A couple of questions about this implementation:

1. Is `after_initialize` called once during book per controller or on every request for the controller being called? I don't remember if rails keeps these around or not.
2. Rather than call `watch_controller_descendants_for` for every controller entry each time could the callback just walk the ancestry tree of the controller that is currently `self` and look to see if an ancestor is in the list?

In terms of a possible simplification, I think the main reason to support inheritance is just for the `instrument_action :all` in `ApplicationController` case. If supporting this generally introduces a bunch of initialization complexity it might make sense to just special-case support that one case?

Any thoughts @bdehamer ?

**Comment by chancefeick** [comment]
> Is after_initialize called once during book per controller or on every request for the controller being called?

My understanding that it is only called once, [after the frameworks are initialized](http://api.rubyonrails.org/classes/Rails/Railtie/Configuration.html#method-i-after_initialize).

> could the callback just walk the ancestry tree of the controller that is currently self and look to see if an ancestor is in the list?

Yeah, that makes sense.

> If supporting this generally introduces a bunch of initialization complexity it might make sense to just special-case support that one case?

My initial attempt was to add a special case for `ApplicationController`  similar to [:all](https://github.com/librato/librato-rails/blob/feature/instrument_action_all/lib/librato/rails/subscribers.rb#L18), but the issue I encountered was that `descendants` were not loaded yet. Definitely open to a different approach if it reduces complexity.

**Comment by asoong** [comment]
Bump. Would love to see this merged in!

**Comment by bdehamer** [comment]
👍

**Comment by nextmat** [comment]
LGTM

**Pull request merged**